shared/clearwater/sticky_component.rb

```diff
@@ -1,79 +1,48 @@
 require 'clearwater/component'
-require 'clearwater/black_box_node'
 
 module Clearwater
   class StickyComponent
     include Clearwater::Component
 
-    attr_accessor :props, :renderer
-    attr_reader :children
-
-    def self.render props=nil, children=nil
-      unless `props === nil || props.$$is_hash`
-        children = props
-        props = nil
+    class << self
+      alias_method :_new, :new
+      def new(*args)
+        Wrapper.new(*args) { |*arguments| _new(*arguments) }
       end
-
-      Wrapper2.new(self, props, children)
     end
 
-    def initialize(props=nil, children=nil)
-      @props = self.class.make_props props
-      @children = children
+    def initialize(*args)
+      update(*args)
     end
 
-    def should_update? next_props
+    def should_update?(*args)
       true
     end
 
-    def component klass, props=nil, children=nil
-      klass.render props, children
-    end
-
-    def will_receive_props props
-    end
-
-    def will_update props
+    def update(*args)
     end
 
-    def did_update old_props
+    def before_update
     end
 
-    def will_mount
+    def after_update
     end
 
-    def did_mount
+    def before_mount
     end
 
-    def will_unmount
+    def after_mount
     end
 
-    def call &block
-      if renderer
-        renderer.update_dom(&block)
-      else
-        Clearwater::Application.render(&block)
-      end
-    end
-
-    def self.make_props props
-      if `props.$$is_hash`
-        Props.new(props)
-      elsif props.nil?
-        Props.new
-      else
-        props
-      end
+    def before_unmount
```

Comment:
Crap, it looks like hooks don't work on thunks — only vnodes. I wrote a quick `UnmountObserver` class for `Wrapper` to use (specifically, `Wrapper` set one up as an instance variable for `virtual-dom` to read as one of the properties of the thunk) and hooks weren't firing. When I replaced …

```
return #{component.render}`
```

… with …

```
return h('div', { observer: #@unmount_observer }, [#{component.render}]);
```

… they began firing.